### PR TITLE
update costs of unknown conditions, part of the hard fork

### PR DESCRIPTION
**Pull request by arvidn**
This extends the condition codes that have cost to all 2-byte conditions (except ones with leading zero).

The formula changes to use a lookup table based on the second byte.

### CODE REVIEW

File: src/gen/opcodes.rs [hunk 1]
```diff
@@ -61,14 +61,39 @@ pub const CREATE_COIN_COST: Cost = 1800000;
...
+    208000000, 221000000, 235000000, 249000000, 265000000, 282000000, 299000000, 318000000,
+    338000000, 359000000, 382000000, 406000000, 431000000, 458000000, 487000000, 517000000,
+];
 
 pub fn compute_unknown_condition_cost(op: ConditionOpcode) -> Cost {
-    match op {
-        256..=511 => 1000000 + ((op & 0x7) as Cost) * 200000,
-        _ => 0,
+    if op < 256 {
+        println!("op: {op}");
```
Comment (richardkiss) [inline src/gen/opcodes.rs:93]: We should probably remove this

File: src/gen/opcodes.rs [hunk 2]
```diff
@@ -61,14 +61,39 @@ pub const CREATE_COIN_COST: Cost = 1800000;
 pub const AGG_SIG_COST: Cost = 1200000;
 
 // when ENABLE_SOFTFORK_CONDITION is enabled
-// 2-byte condition opcodes whose first byte is 1 have costs, despite being unknown:
+// 2-byte condition opcodes have costs according to this table:
 
-// cost = 1000000 + (byte[1] & 0x7) * 200000
+const COSTS: [Cost; 256] = [
```
Comment (richardkiss) [inline src/gen/opcodes.rs:66]:
```suggestion
// the values `100 * (17 ** idx)/(16 ** idx)` rounded to three significant decimal figures
const COSTS: [Cost; 256] = [
```

File: src/gen/opcodes.rs (hunk 1 above)
Comment (arvidn) [reply src/gen/opcodes.rs:93]: oops :)

**Review by richardkiss — APPROVED** [review]
I could try doing a PR to calculate this at run-time, in a portable way.

**Pull request merged**